frontend/components/LocaleMenu.tsx

```diff
@@ -0,0 +1,202 @@
+import { Box, chakra, Checkbox, FormLabel } from "@chakra-ui/react";
+import Link from "next/link";
+import React, { PropsWithChildren, useState } from "react";
+import { MenuContext, useMenuContext } from "./MenuContext";
+
+export function LocaleMenu(props: PropsWithChildren<unknown>) {
+  const [isOpen, setIsOpen] = useState(false);
+  function toggle() {
+    setIsOpen((isOpen) => !isOpen);
+  }
+
```

Comment:
It's probably a good idea to memoize both `toggle` (with `useCallback`) and the context value (with `useMemo`)